src/test/regress/citus_tests/common.py

```diff
@@ -1015,12 +1036,21 @@ def cleanup_publications(self):
                 )
             )
 
-    def cleanup_logical_replication_slots(self):
-        for slot in self.logical_replication_slots:
-            self.sql(
-                "SELECT pg_drop_replication_slot(slot_name) FROM pg_replication_slots WHERE slot_name = %s",
-                (slot,),
-            )
+    def cleanup_replication_slots(self):
+        for slot in self.replication_slots:
+            start = time.time()
+            while True:
```

Comment:
Potential infinite loop. 
You could break/raise after a number of trials maybe.

Reply:
The exception handler contains this, so it should fail after 10 seconds:
 ```python
                    if time.time() < start + 10:
                        time.sleep(0.5)
                        continue
                    raise
```